packages/from-solana/src/SolanaDomainWallet.ts

```diff
@@ -0,0 +1,255 @@
+import {
+  AccountAuthenticator,
+  AccountAuthenticatorAbstraction,
+  AnyRawTransaction,
+  Ed25519Signature,
+  generateSigningMessageForTransaction,
+  hashValues,
+  Network,
+  NetworkToChainId,
+  NetworkToNodeAPI,
+  Signature,
+} from '@aptos-labs/ts-sdk';
+import {
+  AccountInfo,
+  APTOS_CHAINS,
+  AptosConnectOutput,
+  AptosFeatures,
+  AptosSignMessageInput,
+  AptosSignMessageOutput,
+  AptosWallet,
+  NetworkInfo,
+  UserResponse,
+  UserResponseStatus,
+  WalletIcon,
+} from "@aptos-labs/wallet-standard";
+import { WalletError } from '@solana/wallet-adapter-base';
+import { StandardWalletAdapter as SolanaWalletAdapter } from "@solana/wallet-standard-wallet-adapter-base";
+import {
+  makeSolanaSigningMessageFromRawTransaction,
+  makeSolanaSigningMessageFromStructuredMessage,
+  SolanaDomainIdentity,
+  SolanaDomainPublicKey,
+} from './crypto';
+import { encodeStructuredMessage } from './shared';
+
+const authenticationFunction = '0x7::solana::authenticate';
+
+export interface SolanaDomainWalletOptions {
+  domain?: string;
+  defaultNetwork?: Network;
+}
+
+export class SolanaDomainWallet implements AptosWallet {
+  readonly solanaWallet: SolanaWalletAdapter;
+  readonly domain: string;
+  readonly defaultNetwork: Network;
+
+  readonly version = "1.0.0";
+  readonly name: string;
+  readonly icon: WalletIcon;
+  readonly url: string;
+  readonly accounts = [];
+  readonly chains = APTOS_CHAINS;
+
+  constructor(solanaWallet: SolanaWalletAdapter, options: SolanaDomainWalletOptions = {}) {
+    const { domain, defaultNetwork = Network.MAINNET } = options;
+    this.solanaWallet = solanaWallet;
+    this.domain = domain ?? window.origin;
+    this.defaultNetwork = defaultNetwork;
+    this.name = `${solanaWallet.name} (Solana)`;
+    this.icon = solanaWallet.icon;
+    this.url = solanaWallet.url;
+
+    // const eventsFeature = this.solanaWallet.wallet.features['standard:events'];
+    // console.log(this.solanaWallet.name, 'initial value', this.solanaWallet.wallet.accounts);
+    // eventsFeature.on('change', (args) => {
+    //   console.log('changed', args);
+    // });
+    // this.solanaWallet.on('connect', (account) => {
+    //   console.log(this.solanaWallet.name, 'connected', account);
+    // });
+  }
+
+  readonly features: AptosFeatures = {
+    'aptos:connect': {
+      version: '1.0.0',
+      connect: () => this.connect(),
+    },
+    'aptos:disconnect': {
+      version: '1.0.0',
+      disconnect: () => this.disconnect(),
+    },
+    'aptos:account': {
+      version: '1.0.0',
+      account: () => this.getActiveAccount(),
+    },
+    'aptos:onAccountChange': {
+      version: '1.0.0',
+      // Theoretically supported, but not sure how
```

Comment:
can refer to https://github.com/aptos-labs/aptos-wallet-adapter/pull/500/files#diff-907d94e30b45520090d144d8500765e27e7a4bb68a6a5dbcb865ea8ed7fe5d94R147-R164